doc/user-guide/usage.md

```diff
@@ -0,0 +1,380 @@
+
+This guide assumes you followed steps in [Readme][repo-readme] and LVM operator
+(hereafter LVMO) is running in your cluster.
+
+Below are the available disks in our test kubernetes cluster node and there are
+no existing LVM Physical Volumes, Volume Groups and Logical Volumes
+
+``` console
+sh-4.4# lsblk
+NAME    MAJ:MIN RM   SIZE RO TYPE MOUNTPOINT
+sdb       8:16   0 893.8G  0 disk
+|-sdb1    8:17   0     1M  0 part
+|-sdb2    8:18   0   127M  0 part
+|-sdb3    8:19   0   384M  0 part /boot
+`-sdb4    8:20   0 893.3G  0 part /sysroot
+sr0      11:0    1   987M  0 rom
+nvme0n1 259:0    0   1.5T  0 disk
+nvme1n1 259:1    0   1.5T  0 disk
+nvme2n1 259:2    0   1.5T  0 disk
+sh-4.4# pvs
+sh-4.4# vgs
+sh-4.4# lvs
+```
+
+Here LVMO is installed in `lvm-operator-system` namespace via `make deploy`
+target and operations will not change if LVMO is installed in any other namespaces.
+
+``` console
+kubectl get pods
+NAME                                 READY   STATUS    RESTARTS   AGE
+controller-manager-8bf864c85-8zjlp   3/3     Running   0          96s
+```
+
+After all containers in above listing is in `READY` state we can proceed with
+deploying LVMCluster CR
+
+``` yaml
+cat <<EOF | kubectl apply -f -
+apiVersion: lvm.topolvm.io/v1alpha1
+kind: LVMCluster
+metadata:
+  name: lvmcluster-sample
+  # replace the namespace to suit your installation
+  namespace: lvm-operator-system
+spec:
+  deviceClasses:
+  - name: vg1
+EOF
+```
+
+- Ability to respect Node Selector, Tolerations and Device Selectors in above CR
+  is coming soon.
+- After deploying above CR all free disks will be used to create a volume
+  group, topolvm csi is deployed and all the corresponding resources are
+  created, including storage class for ready consumption of Persistent Volumes
+  based on Topolvm
+
+- Wait for all the pods to be in `READY` state
+- `topolvm-controller` and `topolvm-node` will be in `init` stage for a while
+  awaiting creation of certs and creation of volume groups by `vg-manager`
+  respectively
+
+``` console
+kubectl get pods
+NAME                                  READY   STATUS    RESTARTS   AGE
+controller-manager-8bf864c85-8zjlp    3/3     Running   0          31m
+topolvm-controller-694bc98b87-x6dxn   4/4     Running   0          10m
+topolvm-node-pbcth                    4/4     Running   0          10m
+vg-manager-f979f                      1/1     Running   0          10m
+```
+When all pods are ready, the LVM physical volumes and volume groups should be
+created on the host.
+
+``` console
+sh-4.4# pvs
+  PV           VG  Fmt  Attr PSize  PFree 
+  /dev/nvme0n1 vg1 lvm2 a--  <1.46t <1.46t
+  /dev/nvme1n1 vg1 lvm2 a--  <1.46t <1.46t
+  /dev/nvme2n1 vg1 lvm2 a--  <1.46t <1.46t
+sh-4.4# vgs
+  VG  #PV #LV #SN Attr   VSize  VFree 
+  vg1   3   0   0 wz--n- <4.37t <4.37t
+```
+
+Confirm that the following resources have been created on the cluster:
+
+- When LVMCluster CR is deployed which contains details of all volume groups that
```

Comment:
Is there a design doc that explains why the LVMCluster CR embeds the LVMVolumeGroup CRs? This goes against K8s patterns, but is a bigger discussion for the team outside this PR...

Reply:
- we only have bits and pieces in dev-guide
- yes, it indeed need more discussion and at the same time current implementation provides a feasibility to make LVMVolumGroups CR as independent unit if need may arise